compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -804,12 +799,11 @@ public void visit(BLangMatchStatement matchStatement) {
         matchExprType = matchStatement.expr.type;
 
         boolean currentErrorThrown = this.errorThrown;
-        this.hasLastPatternInClause = false;
-        boolean containsLastPatternInStatement = false;
+        this.hasLastPatternInStatement = false;
+        matchStatement.hasLastPattern = false;
```

Comment:
Instead of adding a new field `hasLastPattern` to `BLangMatchStatement`, I think we should be able to extract this information based on just `this.hasLastPatternInStatement`.

Reply:
Yes, refactored that part